transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/AttachmentHandlerImpl.kt

```diff
@@ -177,6 +186,12 @@ internal class AttachmentHandlerImpl(
     }
 }
 
+private fun ByteArray.isInvalid(maxFileSizeKB: Long?): Boolean =
```

Comment:
Why do you check if it is invalid rather than checking if it is valid?

Reply:
Great observation.

The choice to check if the byteArray is invalid rather than valid was influenced by the usage context. Since a failure to validate the file size triggers an exception, the condition reads more intuitively as: "if byteArray is invalid, then throw." It adds a level of clarity to the logic, making it easier to follow (at least on my opinion).
 I find it more readable than using !valid.